### ISSUE TITLE
Cell loses focus when clicking on its outline

### ISSUE THREAD

**Issue by seancolsen** [centerofci/mathesar#2987]
## Steps to reproduce

1. Prerequisites:
    1. On the table page click on a cell to focus it.
    1. When the cell is focused, you can use the <kbd>Up</kbd> and <kbd>Down</kbd> arrow keys to move the active cell. This is good.
    1. Click on the "Filter" button above the table to open the filtering dropdown. Click it again to close it. Now that button is focused instead of a cell, and you won't be able to use the arrow keys to move the active cell. This is okay, but important to understand.

1. Increase your browser zoom to something like 300% to make this easier.
1. Click on a cell to focus it.
1. Click on the blue cell border.
1. Expect the cell to remain focused.
1. Observe that the cell loses focus.

**Comment by manthan-jsharma** [comment]
@pavish , @rajatvijay , is somebody looking into this issue ? 

I'm interested in picking this up

**Comment by sahil-adit** [comment]
I am interested to solve this issue,Can you please provide some more information?

**Comment by seancolsen** [comment]
@sahil-adit you're welcome to work on this. Please submit a PR or ask specific questions. If you have questions about this issue, you can ask them here. Or if you have questions about getting set up with Mathesar development, please ask them in Matrix.

**Comment by abhirupPann** [comment]
Is this issue still open?

**Comment by seancolsen** [comment]
Yes, @abhirupPann this is still open. You're welcome to submit a PR for it.

**Comment by Mahas1234** [comment]
Are there any more fixes needed. I'm looking for first contribution.

**Comment by seancolsen** [comment]
This issue is still open @Mahas1234, feel free to submit a PR.

**Comment by smit9924** [comment]
Is this issue still open? I would like to fix this up

**Comment by Anish9901** [comment]
Yes, this is still open @smit9924. We are preparing for our beta release currently, so some pieces of the app might be broken in the develop branch. Please feel free to reach out on the issue or on one of our [Matrix](https://wiki.mathesar.org/community/matrix/) channels if you need help regarding this issue.

**Comment by prakhargupta-jan** [comment]
Is it ok to work on an issue without it being assigned to you I hope I'll not be stepping on anyone's toes. new to Open Source

**Comment by smit9924** [comment]
@Anish9901 I debug this issue and would like to have some suggestions:
(1) On click the filter button it is common that cell loses the focus and it might be expected as user click the other element out of the table. I don't feel there isn't anything unexpected over there. According to me we should not lock focus on table cell.
Even thought if we would like to implement that then we can add functionality on filter close to shift focus on particular/first cell of the table
(2) On clicking blue border cell shift focus is actually due to border is lies outside the cell. So on click focus is shift to the another cell. This also can be fixed. But what I have observe is focus on cell not looks that much accurate/attractive so one should go the recreating/improving cell focus overall.

Please make your suggestion on above tow points so I can start working accordingly.
Thank you

**Comment by Anish9901** [comment]
@smit9924 Please go ahead with option (2) and open up a PR for it, please note that no design changes are required for this issue, So, please stick to fixing the behavior you described in option (2).

**Comment by Sharath-1517** [comment]
Hello everyone, I may have come up with a possible solution for this issue.

## Current Behaviour
![MOV to GIF](https://github.com/user-attachments/assets/cd20726f-c08b-4bf0-890d-f6d456aba382)

## My possilble fix
![fix](https://github.com/user-attachments/assets/8c1a5df4-a561-4706-9a35-e3a9449893c1)

Before proceeding, I wanted to check: is anyone actively working on this? If not, would it be alright for me to submit a pull request with my proposed changes?

Thanks for your time, and let me know if there's any feedback!

**Comment by Sharath-1517** [comment]
Hey all, I have tried a possible fix for this issue.

## Current behaviour
![MOV to GIF](https://github.com/user-attachments/assets/e8487307-84e6-409e-9640-ca28ebda6571)

## My fix
![fix](https://github.com/user-attachments/assets/b56948f0-2e7c-4737-ab97-626dd0d4b44e)

Before proceeding, I wanted to check: is anyone actively working on this? If not, would it be alright for me to submit a pull request with my proposed changes?

Thanks for your time, and let me know if there's any feedback!

**Comment by seancolsen** [comment]
Please submit a PR @Sharath-1517.

**Comment by Sharath-1517** [comment]
> Please submit a PR @Sharath-1517. 

Alright, thanks!

**Comment by pavish** [comment]
@seancolsen I'm removing the "good first issue" label for this issue.

This is,
1. A design concern that we need to agree upon w.r.t where the border/outline is positioned, and
2. It invites questions such as, if we want the focus to remain while keeping the design as is, should it be a `border` or an element?

This will require us to engage with the contributor a lot more than other "good first issues".

**Comment by radheshpai87** [comment]
@pavish I want to just know to fix this can we just have the cell border inwards currently the border is outside the actual cell so when clicked it is bound to loose focus and go to the other cells or is the cell border intended to be outward and then the fix needs to be done. Just needed a clarity on that.

**Comment by pavish** [comment]
@radheshpai87 We've had considerable changes to the cell selection logic since this issue was created, and this issue is no longer relevant.

As of the time of this comment, clicking on the border (that lies outside the cell) selects and focuses the adjacent cell that the border overlaps and I think that behaviour works good.
